src/coreclr/src/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/NonGCStaticsNode.cs

```diff
@@ -136,13 +136,13 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly)
             if (factory.PreinitializationManager.HasLazyStaticConstructor(_type))
             {
                 int alignmentRequired = Math.Max(_type.NonGCStaticFieldAlignment.AsInt, GetClassConstructorContextAlignment(_type.Context.Target));
-                int classConstructorContextStorageSize = GetClassConstructorContextStorageSize(factory.Target, _type);
+                int classConstructorContextStorageSize = GetClassConstructorContextSize(factory.Target, _type);
                 builder.RequireInitialAlignment(alignmentRequired);
                 
-                Debug.Assert(classConstructorContextStorageSize >= GetClassConstructorContextSize(_type.Context.Target));
+                Debug.Assert(classConstructorContextStorageSize >= GetClassConstructorContextStorageSize(_type.Context.Target, _type));
 
                 // Add padding before the context if alignment forces us to do so
-                builder.EmitZeros(classConstructorContextStorageSize - GetClassConstructorContextSize(_type.Context.Target));
+                builder.EmitZeros(classConstructorContextStorageSize - GetClassConstructorContextStorageSize(_type.Context.Target, _type));
```

Comment:
And this too.

Reply:
Could you undo this change? (This subtraction is always equal to zero if left like this.)